controller/instance_manager_controller.go

```diff
@@ -988,6 +1003,32 @@ func (imc *InstanceManagerController) deleteInstanceManagerPDB(im *longhorn.Inst
 	return nil
 }
 
+func (imc *InstanceManagerController) syncOrphans(im *longhorn.InstanceManager) error {
+	// Instances are live inside the instance manager pod.
+	// Remove the instance orphan CRs when the instance manager pod is deleted or not running.
+	var isInstanceManagerPodDeletedOrNotRunning = false
+	if im.Status.CurrentState != longhorn.InstanceManagerStateRunning {
+		isInstanceManagerPodDeletedOrNotRunning = true
+	} else {
+		switch pod, err := imc.ds.GetPodRO(im.Namespace, im.Name); {
+		case datastore.ErrorIsNotFound(err):
+			isInstanceManagerPodDeletedOrNotRunning = true
+		case err != nil:
+			return errors.Wrapf(err, "failed to check instance manager %v pod state while sync orphans", im.Name)
+		default:
+			isInstanceManagerPodDeletedOrNotRunning = pod == nil
+		}
+	}
```

Comment:
Partially extracted from `areDangerZoneSettingsSyncedToIMPod`, but not share as a method since `areDangerZoneSettingsSyncedToIMPod` needs the pod object for further checks.